twitter_timeline/twitter/views.py

```diff
@@ -39,15 +42,40 @@ def home(request, username=None):
         if username is not None:
             user = get_object_or_404(get_user_model(), username=username)
             form = None
-    tweets = Tweet.objects.filter(user=user)
+    users_following = [rel.following for rel 
+                       in Relationship.objects.filter(follower=user)]
+    tweets = Tweet.objects.filter(Q(user=user) | Q(user__in=users_following))
+    following_profile = request.user.is_following(user)
     return render(request, 'feed.html', {
         'form': form,
         'twitter_profile': user,
-        'tweets': tweets
+        'tweets': tweets,
+        'following_profile': following_profile
     })
 
 
 @login_required()
+@require_POST
+def follow(request):
+    followed = get_object_or_404(
+        get_user_model(), username=request.POST['username'])
+    Relationship.objects.create(follower=request.user, following=followed)    
```

Comment:
Don't create Relationship manually, we already have that logic abstracted in User.follow().